src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -274,9 +274,9 @@ Angle BallPlacementPlayFSM::calculateWallKickoffAngle(const Point &ball_pos,
 void BallPlacementPlayFSM::setupMoveTactics(const Update &event)
 {
     // assign all but one of the robots to line up away from the ball placing robot
-    unsigned int num_move_tactics = event.common.num_tactics - 1;
+    int num_move_tactics = event.common.num_tactics - 1;
 
-    if (num_move_tactics == 0)
+    if (num_move_tactics <= 0)
```

Comment:
here's kind of a funny C++ oddity

this casting works because both types are `int` so 2's complement make sure it underflows in a way that makes it appear as -1 when you reinterpret the bits. but this casting fails if you change the size type because of sign extension:

```
int main() {
    uint32_t x = 0;
    
    int64_t y = x - 1;
    
    std::cout << y; // it returns 4294967295 here

    return 0;
}
```

there's no problem here but I just thought it was interesting to know

Reply:
Interesting. So does the code there decrement the uint32_t (hence the underflow) and then cast?

Reply:
Another interesting fact I learned in one of my classes was that _signed_ integer overflow is actually undefined in C/C++.

Reply:
signed integer overflow kinda makes sense because of that bit at the end messing things up